WPMedia/fusion-news-theme-blocks
# TMEDIA-21 - Loads Ads block dynamically

## Description

Update Gallery block, and lead art block to lazy load `ads-block` in for use within the gallery. This accounts for theme users who have gallery or lead art block installed by do not have the ads-block installed.

## Jira Ticket
- [TMEDIA-21](https://arcpublishing.atlassian.net/browse/TMEDIA-21)

## Acceptance Criteria
1. Update the implementation for PEN-435: Google Ads: Gallery Interstitial CubeBLOCKED so that if the Google Ads block is not included in blocks.json, the feature bundle can still be deployed successfully
    * In addition, the gallery should still appear / behave as it does currently (without the new interstitial ad) 
    * If a customer tries to set Gallery Cube Clicks as a Theme Setting but does not have the Google Ads Block installed, then ideally nothing should happen (the interstitial ad functionality is not expected to work, but ideally the gallery should still work) 
2. If a customer has added the Google Ad Block in their blocks.json and they’ve configured Gallery Cube Clicks as a Theme Setting, then the behavior described in PEN-435: Google Ads: Gallery Interstitial CubeBLOCKED should work as expected
3. Tests and documentation are updated as necessary

## Test Steps

Ensure you have linked engine-sdk by having the env setting `ENGINE_SDK_REPO` set
Checkout the branch from the engine SDK Pull request - https://github.com/WPMedia/engine-theme-sdk/pull/228


1. Checkout this branch `git checkout TMEDIA-21-gallery-block-dynamic-dependency`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/gallery-block,@wpmedia/lead-art-block`
3. Verify that a site with the property `galleryCubeClicks` can see ads in their galleries, and lead art galleries
    * Lead Art Gallery - http://localhost/news/2020/11/04/story-that-has-a-gallery-without-a-basic-promo-image-as-lead-art/?_website=the-sun
    * Gallery - http://localhost/pf/gallery-example/?_website=the-sun
4. Verify that a site without the `galleryCubeClicks` property can still see their gallery
    * Lead Art Gallery - http://localhost/news/2020/11/04/story-that-has-a-gallery-without-a-basic-promo-image-as-lead-art/?_website=the-gazette
    * Gallery - http://localhost/pf/gallery-example/?_website=the-gazette
5. Remove `"@wpmedia/ads-block@canary",` from the blocks array in the `blocks.json` file of fusion repo
6. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/gallery-block,@wpmedia/lead-art-block`
7. Verify that a site with the property `galleryCubeClicks` does not see ad in gallery, but sees a message in place of the ad, and gallery still works as expected
    * Lead Art Gallery - http://localhost/news/2020/11/04/story-that-has-a-gallery-without-a-basic-promo-image-as-lead-art/?_website=the-sun
    * Gallery - http://localhost/pf/gallery-example/?_website=the-sun
8. Verify that a site without the `galleryCubeClicks` property can still see their gallery
    * Lead Art Gallery - http://localhost/news/2020/11/04/story-that-has-a-gallery-without-a-basic-promo-image-as-lead-art/?_website=the-gazette
    * Gallery - http://localhost/pf/gallery-example/?_website=the-gazette


## Effect Of Changes
### After

When you have `galleryCubeClicks` set and no ad-block in the placement of an ad you will see a message "Missing ad block"
<img width="971" alt="TMEDIA-21-missing-text" src="https://user-images.githubusercontent.com/868127/109009121-12621a80-76a6-11eb-9475-b25a4f449967.png">

Working ad placement
<img width="954" alt="TMEDIA-21-add-placement" src="https://user-images.githubusercontent.com/868127/109009309-4b9a8a80-76a6-11eb-8864-5df8e5e74757.png">


## Dependencies or Side Effects
Requires SDK code from PR - https://github.com/WPMedia/engine-theme-sdk/pull/228

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist 
_The reviewer of the PR should copy-paste this template into the review comments on review._
